### PR TITLE
Support project dependencies from included builds in java modules

### PR DESCRIPTION
**Pull request by mark-vieira**
This pull request updates our Gradle plugin for inferring the Java module path to support project dependencies resolved from included builds. Handling project dependencies in composite builds is slightly tricky since you cannot do a simple `findProject()` with a path from an included build project. In fact, there's no API to get a handle to a `Project` for projects from an included build at all. We mainly only need this in order to find the project directory, in order to check for the existance of a `module-info.java` file. We make a best-effort attempt here to infer the project directory based on the project path, with some special handling for exceptions we know of (i.e. `elasticsearch-core` project lives in a folder called `core`).

### CODE REVIEW

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Review by ChrisHegarty — APPROVED** [review]
LGTM.

File: build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/ElasticsearchJavaModulePathPluginFuncTest.groovy [hunk 1]
```diff
@@ -23,6 +25,19 @@ class ElasticsearchJavaModulePathPluginFuncTest extends AbstractJavaGradleFuncTe
 
     public static final String ES_VERSION = VersionProperties.getElasticsearch()
 
+    public static final String COMPILE_JAVA_CONFIG = """
+        tasks.named('compileJava').configure {
+            doLast {
+                def sep = org.gradle.internal.os.OperatingSystem.current().isWindows() ? ':' : ';'
```
Comment (breskeby) [inline build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/ElasticsearchJavaModulePathPluginFuncTest.groovy:31]: We can switch to `org.elasticsearch.gradle.OS` here and avoid using gradle internal api

**Review by breskeby — APPROVED** [review]
minor remark. otherwise lgtm

**Pull request merged**